### ISSUE TITLE
move cursor will block some time

### ISSUE THREAD

**Issue by seventhking** [purcell/emacs.d#357]
when i overview a lua code file which contains 2000 lines data, i used C-M-\ to format the file,then i use command that move cursor, i found it will block some time, and cpu is 100%.
after C-M-\ and saved the file, a highlight column appeared and beside linenumbers,then "block" behaves.

**Comment by purcell** [comment]
It's probably an inefficiency in `lua-mode`. Perhaps try starting `emacs` with `-Q`, then load `lua-mode` manually, then repeat the test on your sample file. If it's still slow, then there's no problem with this emacs config, and you should submit a bug report to the `lua-mode` authors.

**Comment by seventhking** [comment]
hi, @purcell ,i just tried as you said,then i also test by prelude-emacs ,space-emacs and your config without anything added.then i found when under your config, after "C-M-\", the "block" seriously.

**Comment by seventhking** [comment]
oh,no. maybe it is my fault.
the same config in my own computer works well,but the company doesn't. i can't get an answer.

**Comment by seventhking** [comment]
a git project may cause that problem? project in company is git project, but home isn't.

**Comment by purcell** [comment]
Ah. You've enabled `linum-mode`? If so, try `nlinum-mode` instead - it's faster in exactly this case, so perhaps that's the problem here.

**Comment by seventhking** [comment]
@purcell, It's more efficient now.
Thanks

**Comment by purcell** [comment]
Yes, line numbering will always cause a slow-down with large files. But the speed also depends a lot on the efficiency of the major mode, so there might be problems with `lua-mode`. You can actually find this out by using the profiler: set things up so that you can reproduce a slow-down, then run `elp-instrument-package` to instrument all the `lua-` functions. Afterwards, `elp-results` should show you if any of those functions were really slow. This isn't an issue I can help with very much unless you help me to reproduce it, so I am likely to close it.

**Comment by seventhking** [comment]
Hi, Steve. The report that looks like below:
mem-report:
- command-execute                                         955,890,535  79%
 - call-interactively                                     955,890,535  79%
  - apply                                                 955,890,535  79%
   - ad-Advice-call-interactively                         955,890,535  79%
    - #<subr call-interactively>                          955,457,527  79%
     - next-line                                          842,944,830  69%
      - funcall                                           842,940,686  69%
       - #<compiled 0x109686d>                            842,940,686  69%
        - line-move                                       842,940,686  69%
         - line-move-partial                              833,199,624  69%
          - pos-visible-in-window-p                       468,091,348  38%
           - eval                                         467,967,947  38%
            - if                                          467,924,078  38%
             - format                                     467,874,950  38%
              - projectile-project-name                   467,773,728  38%
               - funcall                                  467,746,235  38%
                - #<compiled 0x188007b>                   467,746,235  38%
                 - projectile-project-root                467,746,235  38%
                  - file-truename                         415,761,475  34%
                   - file-truename                        363,718,439  30%
                    - file-truename                       311,726,107  25%
                     - file-truename                      259,455,400  21%
                      - file-truename                     207,521,344  17%
                       - file-truename                    155,595,168  12%
                        - file-truename                   103,697,976   8%
                           file-truename                   51,783,792   4%
                       file-symlink-p                           3,530   0%
                 projectile-default-project-name               24,564   0%
             file-remote-p                                     10,240   0%
          - posn-at-point                                 234,448,050  19%
           - eval                                         234,448,050  19%
            - if                                          234,448,050  19%
             - format                                     234,423,486  19%
              - projectile-project-name                   234,423,486  19%
               - funcall                                  234,390,734  19%
                - #<compiled 0x188007b>                   234,390,734  19%
                 - projectile-project-root                234,390,734  19%
                  - file-truename                         208,439,294  17%
                   - file-truename                        182,470,998  15%
                    - file-truename                       156,279,822  12%
                     - file-truename                      130,260,514  10%
                      - file-truename                     104,268,922   8%
                       - file-truename                     78,305,834   6%
                        - file-truename                    52,398,178   4%
                           file-truename                   26,465,650   2%
                      file-symlink-p                            2,929   0%
                 projectile-default-project-name               32,752   0%
          + scroll-up                                       2,985,912   0%


cpu report:

- indent-guide-post-command-hook                                12857  40%
 - indent-guide-show                                                       12833  40%
    line-number-at-pos                                                       1609   5%
  + indent-guide--active-overlays                                          95   0%
    indent-guide--make-overlay                                             58   0%
  + indent-guide--beginning-of-level                                     36   0%
    back-to-indentation                                                          24   0%
- linum-update-current                                                    12059  38%
 - linum-update                                                                12058  38%
  - mapc                                                                           12055  38%
   - linum-update-window                                                  12022  38%
    - window-end                                                                 3791  12%
     - jit-lock-function                                                           3791  12%
      - jit-lock-fontify-now                                                      3791  12%
       - funcall                                                                      3791  12%
        - #<compiled 0x489f19>                                            3791  12%
         - run-hook-with-args                                                 3791  12%
          - font-lock-fontify-region                                          3777  11%
           - font-lock-default-fontify-region                             3777  11%
            - font-lock-fontify-keywords-region                        3774  11%
             - #<lambda 0x17781b357b69efa9>                            4   0%
              - apply                                                                       4   0%

**Comment by purcell** [comment]
Hmmm. So the cpu report suggests that disabling `indent-guide-mode` and `linum` for those big files would help. Not sure what to conclude from the memory report, sorry.

**Comment by seventhking** [comment]
Thanks a lot.
I shall disable linum mode.
